nixos/modules/services/networking/syncthing.nix

```
@@ -530,6 +520,10 @@ in {
       This option was removed because Syncthing now has the inotify functionality included under the name "fswatcher".
       It can be enabled on a per-folder basis through the web interface.
     '')
+    (mkRenamedOptionModule [ "services" "syncthing" "extraOptions" ] [ "services" "syncthing" "settings" ])
+    (mkRenamedOptionModule [ "services" "syncthing" "folders" ] [ "services" "syncthing" "settings" "folders" ])
```

Something like this (along with the other options):

```suggestion
    (mkRenamedOptionModule [ "services" "syncthing" "folders" ] [ "services" "syncthing" "settings" "folders" ])
    (mkRenamedOptionModule [ "services" "syncthing" "folders" "rescanIntervalS" ] [ "services" "syncthing" "settings" "folders" "rescanInterval" ])
```

---

hmm, this gives me an eval error. not sure how we can fix that

---

Because the folder's path is supposed to be in between the `"folders"` and the `rescanIntervalS`... I'm not sure how to do that, as the path can be anything of course.

---

I tend to think it's impossible to perform such a 'variable' rename. If indeed we won't find a way to do it, I think it'd be OK to give up on this, as anyway we rename an module option a level higher, which might trigger hopefully a second look on the documentation anyway..

Also, in anycase, it'd be nice to mention all of the changes here in the release notes.. (@Xyz00777 )

---

> ...so the solution we went with here is really "silently ignore the problem and hope the user will read the release notes"? How is this acceptable?
> 
> We should either reintroduce the mapping or add an assertion.

Indeed I forgot about the release notes entry... If you know how to create that mapping, or add an assertion, I'd be happy to review and test it. I am not sure it's possible to do it TBH. I also asked for help on discourse and got no response.

---

I've added an assertion in https://github.com/NixOS/nixpkgs/pull/232439. We can't use the module system for this because of https://github.com/NixOS/nixpkgs/issues/96006.

Could you open a PR for the release note entry?